calicoctl/calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -100,19 +103,26 @@ type resourceHelper struct {
 func (rh resourceHelper) String() string {
 	if !rh.isList {
 		return fmt.Sprintf("Resource(%s %s)", rh.resource.GetObjectKind(), rh.resource.GetObjectKind().GroupVersionKind())
-
 	}
 	return fmt.Sprintf("Resource(%s %s)", rh.listResource.GetObjectKind(), rh.listResource.GetListMeta().GetResourceVersion())
 }
 
 // Store a resourceHelper for each resource.
-var helpers map[schema.GroupVersionKind]resourceHelper
-var kindToRes = make(map[string]ResourceObject)
+var (
+	helpers   map[schema.GroupVersionKind]resourceHelper
+	kindToRes = make(map[string]ResourceObject)
+	allKinds  = []string{}
+)
+
+func ValidResources() []string {
+	sort.Strings(allKinds)
+	return allKinds
+}
```

Comment:
What do you think about adding ValidResourcesList(), that returns the already sorted and formatted list

`
func ValidResourcesList() string {
  kinds := sort.String(allKinds)
	resourceList := ""
	for _, r := range kinds {
		resourceList += fmt.Sprintf("    - %s\n", r)
	}
	return resourceList
}
`
Then if we ever want to change the format, we can do it in one place. Maybe a better place for this would be https://github.com/projectcalico/calico/blob/master/calicoctl/calicoctl/util/docstring.go